lib/python/picongpu/plugins/plot_mpl/energy_waterfall_visualizer.py

```diff
@@ -0,0 +1,169 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2018 PIConGPU contributors
+Authors: Sophie Rudat
+License: GPLv3+
+"""
+
+from picongpu.plugins.energy_histogram import EnergyHistogram
+from picongpu.plugins.plot_mpl.base_visualizer import Visualizer as\
+    BaseVisualizer, plt
+import numpy as np
+from matplotlib.colors import LogNorm
+
+
+class Visualizer(BaseVisualizer):
+    """
+    Class for creation of histogram plots on a logscaled y-axis.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        run_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        super(Visualizer, self).__init__(run_directory)
+        self.cbar = None
+
+    def _create_data_reader(self, run_directory):
+        """
+        Implementation of base class function.
+        """
+        return EnergyHistogram(run_directory)
+
+    def _create_plt_obj(self, ax):
+        """
+        Implementation of base class function.
+        Turns 'self.plt_obj' into a matplotlib.pyplot.plot object.
+        """
+        counts, bins, iteration = self.data
+        np_data = np.zeros((len(bins), len(iteration)))
+        for index, ts in enumerate(iteration):
+            np_data[:, index] = counts[ts]
+        max_iter = max(iteration * 1.39e-16 * 1.e12)
```

Comment:
magic numbers for conversion to ps and hard-coded `dt`: as above, replace with `FindTime` module to make more general please :)